HTMLSerializer.js

```diff
@@ -112,6 +126,30 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Takes an html node of type Node.TEXT_NODE, and add its text content with
+   *     all characters properly escaped to |this.html|.
+   * @param {Node} node The text node.
+   */
+   // TODO(sfine): Take care of attribute value normalization:
+   // https://developers.whatwg.org/the-iframe-element.html#the-iframe-element
+  processText(node) {
+    var win = node.ownerDocument.defaultView;
+    var windowDepth = this.windowDepth(win);
+    var text = node.textContent;
+    // & must be processed first.
```

Comment:
It's best to use comments to answer "why?" so something like this may be more helpful:
// Some escaping introduces '&' characters so we escape '&' first to prevent
// escaping the '&' added by other escape substitutions.